addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -142,35 +139,96 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+                    checkWildcardSources(allowedWildcardSources, msg, csp, false);
                 }
 
                 PolicyInOrigin p = new PolicyInOrigin(policy, URI.parseURI(RAND_FQDN).orElse(null));
                 if (p.allowsUnsafeInlineScript()) {
-                    buildScriptUnsafeInlineAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildScriptUnsafeInlineAlert(
+                                    getHeaderField(msg, CspUtils.HEADER_CSP).get(0), csp)
                             .raise();
                 }
 
                 if (p.allowsUnsafeInlineStyle()) {
-                    buildStyleUnsafeInlineAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildStyleUnsafeInlineAlert(
+                                    getHeaderField(msg, CspUtils.HEADER_CSP).get(0), csp)
                             .raise();
                 }
 
                 if (allowsUnsafeHashes(policy, FetchDirectiveKind.ScriptSrc)) {
-                    buildScriptUnsafeHashAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildScriptUnsafeHashAlert(getHeaderField(msg, CspUtils.HEADER_CSP).get(0), csp)
                             .raise();
                 }
 
                 if (allowsUnsafeHashes(policy, FetchDirectiveKind.StyleSrc)) {
-                    buildStyleUnsafeHashAlert(getHeaderField(msg, HTTP_HEADER_CSP).get(0), csp)
+                    buildStyleUnsafeHashAlert(getHeaderField(msg, CspUtils.HEADER_CSP).get(0), csp)
                             .raise();
                 }
             }
         }
+        if (CspUtils.hasMetaCsp(source)) {
+
+            checkMetaPolicy(msg, id, source, true, cspHeaderFound);
+        }
 
         LOGGER.debug("\tScan of record {} took {} ms", id, System.currentTimeMillis() - start);
     }
 
+    private void checkMetaPolicy(
+            HttpMessage msg, int id, Source source, boolean isMeta, boolean hasHeader) {
+        List<Element> cspMetaElements = getMetaPolicies(source);
+        List<PolicyError> observedErrors = new ArrayList<>();
+        if (!cspMetaElements.isEmpty()) {
+            for (Element element : cspMetaElements) {
+                String metaField = element.getAttributeValue("http-equiv");
+                String metaPolicy = element.getAttributeValue("content");
+                if (CspUtils.HEADER_CSP.equalsIgnoreCase(metaField)
+                        && !StringUtils.isBlank(metaPolicy)) {
+                    List<PolicyError> metaObservedErrors = new ArrayList<>();
+                    PolicyErrorConsumer metaConsumer =
+                            (severity, message, directiveIndex, valueIndex) -> {
+                                // Skip notices for directives that Salvation doesn't
+                                // handle
+                                if (ALLOWED_DIRECTIVES.stream().noneMatch(message::contains)) {
+                                    observedErrors.add(
```

Comment:
Was this meant to be `metaObservedErrors`? Otherwise this does not seem to have any effect.